jobs/build/ocp/Jenkinsfile

```diff
@@ -162,6 +162,58 @@ def get_mirror_url(build_mode, version) {
     return "https://mirror.openshift.com/enterprise/enterprise-${version}"
 }
 
+def get_changelog(rpm_name, record_log) {
+    //
+    // INPUTS:
+    //   rpm_name - the name of an RPM build previously
+    //   record_log - an array of build records with | separated fields
+
+    rpm_builds = record_log['build_rpm']
+    if (rpm_builds == null || rpm_builds.size() == 0) {
+        return ""
+    }
+
+    // filter for the desired RPM using name
+    build_record_index = rpm_builds.findIndexOf {
+        it['rpm'] == rpm_name
+    }
+    if (build_record_index == -1) {
+        return ""
+    }
+    build_record = rpm_builds[build_record_index]
+
+    // then get the task_id and task_url out of it
+    // task_id = build_record['task_id']
+    task_url = build_record['task_url']
+
+    // get the build ID from the web page
+    // there must be an API way to do this MAL 20180622
+    build_id = sh(
```

Comment:
This should really silently fail or retry. I give 100% odds that a build will "Fail" because this curl fails at some point.
Honestly the entire maill_success stuff should be wrapped in a giant try/catch/pass because failing to send out the success email should never, ever give us red.

Reply:
Hrrm.  I think that might be the next little refactor.